source/server/overload_manager_impl.cc

```diff
@@ -22,6 +22,54 @@
 namespace Envoy {
 namespace Server {
 
+/**
```

Comment:
This class declaration was moved from the anonymous namespace section so that we could predeclare it in the header and use it for templatizing the TypedSlot. The class definition was otherwise unchanged.